core/templates/services/exploration-html-formatter.service.ts

```diff
@@ -162,9 +162,14 @@ export class ExplorationHtmlFormatterService {
 
   getAnswerHtml(
       answer: string,
-      interactionId: string,
+      interactionId: string | null,
       interactionCustomizationArgs: InteractionCustomizationArgs
   ): string {
+    // TODO(#14464): Remove this check once interaction ID is
+    // not allowed to be null.
+    if (interactionId === null) {
+      throw new Error('InteractionId cannot be null');
+    }
```

Comment:
Does this error show up in the users browser?

Reply:
Hi, ideally, we'd never want the interactionId to be null. However, in the existing codebase, it is configured to be potentially null in a few places while in others it isn't. When I started to add configure strict typechecks, it complained about this fact. Therefore, this error throwing is added as a temporary check till the time we don't fix #14464, and I do not anticipate this will actually be used. 

I discussed this with Vojtěch as well, but, do you propose any other way of handling this?

Reply:
I wanted to check when exactly it is null and what user actions will result in this error. I ask this because if it is a common scenario, it will end up throwing a bunch of errors and all of them will get recorded in our logs (this may end up hiding more serious errors).

Reply:
I found that interaction Id is actually Interaction type e.g. NumericInput` which can never be null from the UI (specified in the ExplorationEditorMainTab.js file). But it looks like interactionId depends a lot on the context, sometimes fetched from backend, sometimes fetched from frontend. Since I am pretty new at the moment (with limited understanding of the codebase), I am not exactly sure if I will be able to find and verify the usage of interactionId in every single place, and what would probably be easier is to fix the actual bug I filed (i.e. to make sure interactionId is never null) but I believe that's a seperate task altogether. Is there any other way you'd recommend me to handle this instead?

Reply:
Hm, can we make sure that we call this method only when the arguments are properly defined? Ideally, the caller should ensure that valid arguments are passed to the method.

Reply:
So, this function is called from a bunch of places, do you propose that we add null checks for interactionId in each of those places instead?

Reply:
By the way, this PR is exact replica of the earlier one I closed (which messed up a bit due to git history). There was some discussion (https://github.com/oppia/oppia/pull/14417#discussion_r767634035) there as well, and this way of handling it came out to be the recommended way after our discussion, since it seemed better that we added null check in a single place rather than in multiple places.

Reply:
I did a quick check, it looks like interactionId is null when no interaction is set (e.g. when you create a state and not add an interaction). The "getAnswerHtml" methods will never be called in this case because there is no way to specify an "answer" when there is no interaction. Which means this error will never be thrown under normal operation. So I think it's ok to keep this. Thanks for the discussion!